src/core/lib/security/credentials/oauth2/oauth2_credentials.cc

```diff
@@ -72,6 +72,9 @@ grpc_auth_refresh_token grpc_auth_refresh_token_create_from_json(
   }
   result.type = GRPC_AUTH_JSON_TYPE_AUTHORIZED_USER;
 
+  // quota_project_id is optional, so we don't check the result of the copy.
+  grpc_copy_json_string_property(json, "quota_project_id",
```

Comment:
Related to below comment in test.

I've seen some docs that say that this can be a string or a numberic project ID. Does the format used for the refresh token credentials only allow the human readable project name format? If number format is also a possibility, then parsing may need adjusting.

Reply:
@broady Can you confirm?

However, do you think it is problematic if we treat a numeric ID as a string?

Reply:
Answered below. It's always a string. Project number will not be in this field.